Khan/wonder-blocks
# 2/4 [feature:popover] WB-604: Implement PopoverContent

## Summary

Renders the content for the default variants: Text, Icon and Illustration.

### NOTE: Stacked PRs [2/4]

This is part of [WB-579: Implement Popovers](https://khanacademy.atlassian.net/browse/WB-579)

These PRs are created to cover the whole Popover implementation:

1. Implement PopoverContentCore (current PR): Renders popover content.
2. **Implement PopoverContent: Add text-only, icon and illustration variations.**
3. Implement Popover: Adds Popover component to handle popover triggering and positioning. Also, integrates TooltipTail and TooltipPopper.
4. Popover: Add Focus management

## Implementation

### Popover
[Implementation Spec](https://khanacademy.atlassian.net/wiki/spaces/FRONTEND/pages/115310606/Popover+v1+Implementation+Spec#PopoverContentCore-public-)
- Added `PopoverContent` component to display predefined variants inside `PopoverContentCore`.
- Enforced `PopoverContent` constraints using flow (e.g. Disallows to use `emphasized` and `icon` at the same time).

## Testplan
1. Open https://deploy-preview-434--wonder-blocks.netlify.com/#!/PopoverContent
2. Check all the variations are being displayed as expected.
![popover-content](https://user-images.githubusercontent.com/843075/59958425-84677280-9474-11e9-8479-d9c6fb40a1c3.png)
